job_runner/reporting/statemachine.py

```diff
@@ -0,0 +1,60 @@
+import logging
+from typing import Callable, Dict, Tuple, Type
+
+from job_runner.reporting.message import (
+    Message,
+    Exited,
+    Finish,
+    Init,
+    Running,
+    Start,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class TransitionError(ValueError):
+    pass
+
+
+class StateMachine:
+    def __init__(self) -> None:
+
+        logger.debug("Initializing state machines")
+        initialized = (Init,)
+        jobs = (Start, Running, Exited)
+        finished = (Finish,)
+        self._handler: Dict[Message, Callable[[Message], None]] = {}
+        self._transitions = {
```

Comment:
This requires some comments explaining semantics of transitions. For isnstance, what `None` represents?